test/gtest/reducecalculation.hpp

```diff
@@ -167,14 +160,30 @@ struct ReduceCalculationTest : public ::testing::TestWithParam<ReduceCalculation
         auto&& handle            = get_handle();
         reducecalculation_config = GetParam();
         auto gen_value = [](auto...) { return prng::gen_descreet_uniform_sign<T>(1e-2, 100); };
+        auto gen_between_values = [](auto...) { return prng::gen_A_to_B<int>(0, 3); };
 
         dim                 = reducecalculation_config.dim;
         nanPropagation      = reducecalculation_config.nanPropagation;
         reduceCalculationOp = reducecalculation_config.reduceCalculationOp;
 
-        auto in_dims = reducecalculation_config.GetInput();
+        if((reduceCalculationOp == MIOPEN_REDUCE_CALCULATION_SUM ||
+            reduceCalculationOp == MIOPEN_REDUCE_CALCULATION_PROD) &&
+           !(std::is_same<T, float>::value || std::is_same<T, half_float::half>::value ||
+             std::is_same<T, bfloat16>::value))
+        {
+            GTEST_SKIP() << "Unsupported input dtype setup: Input type should be float, half or "
+                            "bfloat16 for numeric calculation.";
+        }
```

Comment:
this maybe unimportant conditions when writing test

Reply:
yes, this is actually unnecessary because input test cases are customized, so you can intentionally avoid those test cases. I removed this